### ISSUE TITLE
CORS Policy issue 

### ISSUE THREAD

**Issue by ahyland** [SharePoint/sp-dev-docs#4734]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior
I enabled the public cdn a number of weeks back on a tenant and a custom spfx web part worked fine for a number of weeks. The custom spfx web part was a simple iframe web part that accepted a number of parameters. 

#### Observed Behavior

Currently when trying to access the app all users are getting a "Something went wrong" error, further inspection in the console is showing:

![image](https://user-images.githubusercontent.com/4320762/66466376-7c126200-ea7a-11e9-8bfb-3f9b2108b40a.png)

The text reads:

> Access to fetch at 'https://publiccdn.sharepointonline.com/tenant.sharepoint.com/sites/apps/ClientSideAssets/c84c0b24-69f9-4b4e-b269-7e376797a518/spfxtoolbox-embedappwebpartstrings_en-us_f4dd924030c898f0b29ea4353bcfe84a.js' from origin 'https://tenant.sharepoint.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

This was first observed this morning and is happening for all users, I am currently logged in as a site collection administrator and observing the issue. 

Appreciate your help.

**Comment by ahyland** [comment]
We had a very similar issue arise again today, but with the following error:

> [SPLoaderError.loadComponentError]:
> ***Failed to load component "f5a91db7-66f4-4521-b0cb-f06e085547e1" (EmbedAppWebPart). Original error: ***Failed to load URL 'https://publiccdn.sharepointonline.com/tenant.sharepoint.com/sites/apps/ClientSideAssets/{GUID}/webpartname-embedappwebpartstrings_en-us_f4dd924030c898f0b29ea4353bcfe84a.js' for resource 'EmbedAppWebPartStrings' in component 'f5a91db7-66f4-4521-b0cb-f06e085547e1' (EmbedAppWebPart). There was a network problem. This may be a problem with a HTTPS certificate. Make sure you have the right certificate.

This issue occurred across all of the site collections for all users at different locations. Appreciate your help on this matter.

**Comment by andrewconnell** [comment]
Is this repeatable or intermittent? Does it happen on all your solutions or just one solution? Does it happen for all users or just a few? Does it happen in one tenant or multiple?

**Comment by ahyland** [comment]
Issue has been resolved, a policy to allow for JS files was missing. Enabling JS files resolved the issue. 

Thanks!

**Comment by andrewconnell** [comment]
Curious... a policy on YOUR environment or within SharePoint Online?

**Comment by florianwachter** [comment]
> Issue has been resolved, a policy to allow for JS files was missing. Enabling JS files resolved the issue.
> Thanks!

@ahyland: Can you please provide more details. I think I have a similar issue and I'm really interested how you solved the issue in your environment. 

Thanks

**Comment by ahyland** [comment]
@andrewconnell it seems to have just been ring fenced to the one environment, however I am yet to enable the CDN on any other environments. 

@florianwachter I used the O365 CLI to list the CDN Policies, I first ran:

`o365 spo cdn policy list -t Public `

The **Include File extensions** showed the following:

> IncludeFileExtensions                 CSS,PNG,ICO

For some reason JS was missing as a file extension, we used the CLI again to run the following:

`o365 spo cdn policy set -t Public -p IncludeFileExtensions -v "CSS,PNG,ICO,JS"`

After running this and enabling the file extensions the custom spfx web part then worked as expected. 

Previously we where disabling the CDN then enabling it again, we found the web part worked for 12-24 hours before the issue reoccurred. 

I am not actually sure how we come to not have the JS missing from the file extension policy. I had previously removed the master page origin using the CLI. 

Hope this helps anyone else who may have this issue.
